clouds/aws/create_aws_network_resources.py

```diff
@@ -0,0 +1,793 @@
+    Assign 'Name' tag value to EC2 resource
+
+    Args:
+         ec2_client (botocore.client.EC2): aws botocore client
+         resource_id (str): The ID of resource(which can be of 'vpc', 'subnet', etc)
+         tag_name (str): The value of 'Name' tag
+    """
+    ec2_client.create_tags(
+        Resources=[
+            res_id,
+        ],
+        Tags=[
+            {
+                "Key": "Name",
+                "Value": tag_name,
+            },
+        ],
+    )
+
+
+def get_subnets_cidr(cidr_block, subnet_count) -> list:
+    """
+    Calculates Subnet CIDR address blocks within given VPC CIDR block
+
+    Args:
+         ec2_client (botocore.client.EC2): aws botocore client
+         cidr_block (str): The IPv4 network of VPC
+         subnet_count (str): The number of additional bits with which to extend the prefix
+
+    Returns:
+         list: The list of subnet CIDR address blocks
+
+    Raises:
+         AddressValueError: If cidr_block provided in invalid for IPv4 address
+    """
+    subnet_range = []
+    net = IPv4Network(address=cidr_block)
+    prefix = net.prefixlen + (subnet_count)
+    for subnet in net.subnets(new_prefix=prefix):
+        subnet_range.append(str(subnet))
+    return subnet_range
+
+
+def get_availability_zones(ec2_client) -> list:
+    """
+    Provides a list of zone IDs in availability zone of current AWS Region
+
+    Args:
+         ec2_client (botocore.client.EC2): aws botocore client
+
+    Returns:
+         list: The list of Zone IDs
+    """
+    zones = []
+    av_zones = ec2_client.describe_availability_zones()
+    for zone in av_zones.get("AvailabilityZones"):
+        zones.append(zone.get("ZoneId"))
+    LOGGER.info(f"Availability zones : { zones }")
+    return zones
+
+
+def create_network_resources(
+    ec2_client, cidr_block, site_name, subnet_count, region_name
+) -> dict:
+    """
+    Creates VPC and other resources such as VPC Endpoint, Public and Private Subnets, Route Tables,
+    Route, Nat gateways, Internet Gateways
+
+    Args:
+         ec2_client (botocore.client.EC2): aws botocore client
+         cidr_block (str): The IPv4 network for creating VPC
+         site_name (str): The value for 'Name' tag used for all resource
+         subnet_count (str): The number of public and private subnet pairs to create
+
+    Returns:
+         dict: The dictionary contains VPC ID, public and private subnet IDs
+
+
+    """
+    verify_aws_credentials()
+    verify_aws_config()
+
+    subnet_cidr_blocks = get_subnets_cidr(
+        cidr_block=cidr_block, subnet_count=subnet_count
+    )
+    availability_zones = get_availability_zones(ec2_client=ec2_client)
+
+    # Check the provided number of public and private subnet pairs should not be more than 3
+    # for creating one public and private subnet per AZ
+    if subnet_count > 3:
+        raise Exception(
+            f"Number of private and public subnet pairs should be less than 3, subnet count provided is {subnet_count}"
```

Comment:
In your error you have `should be less than 3` but you check `subnet_count > 3:`
in this case `subnet_count == 3` will be ok and do not match your error

Reply:
error message updated